Rakefile

```diff
@@ -5,13 +5,42 @@ require 'rspec/core/rake_task'
 
 RuboCop::RakeTask.new
 
+# rubocop:disable Metrics/BlockLength
 namespace :smithy do
-  RSpec::Core::RakeTask.new do |t|
+  RSpec::Core::RakeTask.new('spec:unit') do |t|
     t.pattern = 'gems/smithy/spec/**/*_spec.rb'
     t.ruby_opts = '-I gems/smithy/spec'
     t.rspec_opts = '--format documentation'
   end
+
+  task 'spec:endpoints' do
+    require_relative 'gems/smithy/spec/spec_helper'
+
+    spec_paths = []
+    include_paths = []
+    tmp_dirs = []
+    Dir.glob('gems/smithy/spec/fixtures/endpoints/*/model.json') do |model_path|
+      test_name = model_path.split('/')[-2]
```

Comment:
Is there a better way to express this one?

Reply:
Do you mean the `model_path.split('/')[-2]` part? I could do `model_path.split('/').last(2).first` instead, but that felt worse.  Alternatively we coudl use regex to extract it, but I think splitting on / is simpler here.

Reply:
Was thinking in terms of readability. What exactly is extracted? It's otherwise fine.